### ISSUE TITLE
React Native 0.59, iOS 14.5 and Realm

### ISSUE THREAD

**Issue by SchwarzerSchnee** [realm/realm-js#3580]
As pointed out in https://github.com/realm/realm-js/issues/3557 , upgrading Realm to 10.2.0 fixes the app-start crashes on iOS 14.5 when using a newer RN and pods. This is working just fine for one of the live apps we are maintaining. However, we have another one which is still at RN 0.59, doesn't use pods and it's not feasible to upgrade the whole thing due to various breaking changes. So, I tried to just upgrade Realm so that the app wouldn't crash anymore. Well, turns out now it doesn't even build anymore! It looks like this is fixable, but I don't know how. Any help there?

## Expected Results

Our legacy app that uses Realm runs smoothly on iOS 14.5.

## Actual Results

Old version of Realm: app crashes.
Realm 10.2.0: app doesn't build. Screenshots will be shown in next section.

## Steps to Reproduce

```
react-native init --version="react-native@0.59" realmtest
cd realmtest
npm install realm@10.2.0 --save
react-native link
```

Having done that, open .xcodeproj and build. Build will fail with this:

<img width="848" alt="Screenshot 2021-02-11 at 13 00 53" src="https://user-images.githubusercontent.com/31659632/107635355-49c6d500-6c6b-11eb-97fe-a885cccb333d.png">

With a little digging, I figured that we can get rid of this error by editing the header search paths of the contained RealmReact.xcodeproj to search the src directory (second from top) recursively (since it doesn't by default), like so:

<img width="1218" alt="Screenshot 2021-02-11 at 13 25 26" src="https://user-images.githubusercontent.com/31659632/107636282-b393ae80-6c6c-11eb-92d0-5936eef80a77.png">

But then, a different build error occurs. Now what?

<img width="857" alt="Screenshot 2021-02-11 at 13 27 43" src="https://user-images.githubusercontent.com/31659632/107636488-03727580-6c6d-11eb-8a1c-cfa71b36b8b4.png">

## Version of Realm and Tooling

- Realm JS SDK Version: 10.2.0
- Node or React Native: 0.59
- Client OS & Version: iOS 14.5 beta
- Which debugger for React Native: ?/None

**Comment by praneeth-hiver** [comment]
Facing the same issue, any update on this?

**Comment by steffenagger** [comment]
Hello @SchwarzerSchnee & @praneeth-hiver 

I completely understand not being able to invest the time needed in updating RN for existing project, especially from pre `v0.60`.
Updating to v10, has worked in some use cases, as suggested in #3557 - as you mentioned.

In this case I would suggest rolling back the v10 updates, and try out our newly released [v6.1.6](https://www.npmjs.com/package/realm/v/6.1.6) where the issue have been addressed.

Appreciate any feedback on the fix, and/or other problems encountered with the new xcode/ios beta.

**Comment by marcosrdz** [comment]
> Hello @SchwarzerSchnee & @praneeth-hiver
> 
> I completely understand not being able to invest the time needed in updating RN for existing project, especially from pre `v0.60`.
> Updating to v10, has worked in some use cases, as suggested in #3557 - as you mentioned.
> 
> In this case I would suggest rolling back the v10 updates, and try out our newly released [v6.1.6](https://www.npmjs.com/package/realm/v/6.1.6) where the issue have been addressed.
> 
> Appreciate any feedback on the fix, and/or other problems encountered with the new xcode/ios beta.

Still no librealm-parser-maccatalyst.a in this release. Cant be used with mac catalyst.

**Comment by takameyer** [comment]
Catalyst support is being handled by this issue #2883
We will not be supporting old versions of React Native.
